src/local-state/__tests__/LocalState/types.test.ts

```diff
@@ -0,0 +1,305 @@
+
+import type { ContextValue } from "./fixtures/context-value.js";
+
+type SetRequired<T, Keys extends keyof T> = { [K in Keys]-?: T[K] } & Omit<
+  T,
+  Keys
+>;
+
+describe.skip("Type tests", () => {
+  test("allows resolvers of anything with no generic", () => {
+    interface Food {
+      __typename: "Food";
+      name: string;
+    }
+    new LocalState();
+    new LocalState({ resolvers: {} });
+
+    new LocalState({
+      resolvers: {
+        Query: {
+          currentUserId: () => "1",
+        },
+        User: {
+          favoriteFood: () => ({ __typename: "Food" }),
+        },
+        Food: {
+          name: (parent: Food) => parent.name,
+          ingredients: () => {},
+        },
+      },
+    });
+
+    new LocalState({
+      context: () => ({
+        env: "development",
+      }),
+      resolvers: {
+        Query: {
+          currentUserId: () => "1",
+        },
+        User: {
+          favoriteFood: () => ({ __typename: "Food" }),
+        },
+        Food: {
+          name: (parent: Food) => parent.name,
+          ingredients: () => {},
+        },
+      },
+    });
+  });
+
+  test("works with codegen resolver types", async () => {
+    type Resolvers =
+      import("./fixtures/local-resolvers-without-context-value.js").Resolvers;
+    const { FoodCategory } = await import(
+      "./fixtures/local-resolvers-without-context-value.js"
+    );
+
+    type RequiredRootResolver = SetRequired<Resolvers, "Query">;
+
+    new LocalState<Resolvers>();
+    // @ts-expect-error missing argument
+    new LocalState<RequiredRootResolver>();
+
+    new LocalState<Resolvers>({});
+    // @ts-expect-error missing resolvers option
+    new LocalState<RequiredRootResolver>({});
+
+    new LocalState<Resolvers>({
+      resolvers: {},
+    });
+    new LocalState<RequiredRootResolver>({
+      // @ts-expect-error missing Query resolver
+      resolvers: {},
+    });
+
+    new LocalState<Resolvers>({
+      resolvers: {
+        Query: {
+          currentUserId: () => "1",
+        },
+        User: {
+          favoriteFood: () => ({
+            __typename: "Food",
+            name: "Pasta",
+            categories: [FoodCategory.Italian],
+          }),
+        },
+        Food: {
+          name: (food) => food.name?.toUpperCase() ?? null,
+          categories: (food, { limit, offset }) => {
+            limit = limit ?? 5;
+            return food.categories?.slice(offset, offset + limit) ?? [];
+          },
+        },
+      },
+    });
+    new LocalState<RequiredRootResolver>({
+      resolvers: {
+        Query: {
+          currentUserId: () => "1",
+        },
+        User: {
+          favoriteFood: () => ({
+            __typename: "Food",
+            name: "Pasta",
+            categories: [FoodCategory.Italian],
+          }),
+        },
+        Food: {
+          name: (food) => food.name?.toUpperCase() ?? null,
+          categories: (food, { limit, offset }) => {
```

Comment:
We should test the argument types from `Resolvers` in a few places


```suggestion
            expectTypeOf(food).toEqualTypeOf<Food>();
            expectTypeOf(limit).toEqualTypeOf<Maybe<number> | undefined>();
            expectTypeOf(offset).toEqualTypeOf<Maybe<number> | undefined>();
```

Reply:
Good call! Updated in [8b2a8c3](https://github.com/apollographql/apollo-client/pull/12614/commits/8b2a8c37bcba3e551ec01f386c346bb695f115ba)